ucca/validation.py

```diff
@@ -122,6 +124,106 @@ def validate_non_terminal(self, linkage=False, multigraph=False):
                 if len(edges) > 1:
                     yield "Multiple edges from %s to %s: %s" % (self.node_id, child_id, ", ".join(
                         "%d %s" % (len(e), t) for t, e in tag_to_edge(edges).items()))
+        no_sub_non_scene = (ETags.Process, ETags.Adverbial, ETags.Linker, ETags.Time,
+                            ETags.Quantifier, ETags.Connector, ETags.State)
+        if any(k in self.incoming for k in no_sub_non_scene):
+            s = self.outgoing_tags.difference((set.union({ETags.Terminal, ETags.Punctuation}, NON_SCENE, SUPP_FUNC)))
+            edges_to_check =  [k for k in self.incoming if k in no_sub_non_scene]
+            if s:
+                yield "%s unit (%s) with %s children: %s" % (join(edges_to_check), self.node_id, join(s), self.node)
+        if ETags.Unanalyzable in self.incoming:
+            if self.outgoing:
+                yield "%s unit (%s) with children: %s" % (ETags.Unanalyzable, self.node_id, self.node) # FIXME: should include punctuation and terminal?
+        if ETags.Ground in self.incoming:
+            s = {k for k in {ETags.ParallelScene, ETags.Linker, ETags.Ground} if k in self.outgoing_tags}
+            if s:
+                yield "%s unit (%s) with %s children: %s" % (ETags.Ground, self.node_id, join(s), self.node)
```

Comment:
This `yield` seems identical to the one below (and above) but just the parameters are different. Can it be refactored to avoid repetition? Maybe to a function or by iterating over a data structure containing the rules to check